pkg/eventprocessor/impl/executor/impl/blockscope.go

```diff
@@ -215,6 +216,84 @@ func (bs *blockScope) StateHash(ctx context.Context, chainID tableland.ChainID)
 	return executor.NewStateHash(chainID, bs.scopeVars.BlockNumber, hash), nil
 }
 
+func (bs *blockScope) CalculateTreeLeaves(ctx context.Context, chainID tableland.ChainID) error {
+	rows, err := bs.txn.QueryContext(ctx, "select prefix, id from registry where chain_id = ?1 ORDER BY rowid", chainID)
+	if err != nil {
+		return fmt.Errorf("fetching tables from registry: %s", err)
+	}
+	defer func() {
+		_ = rows.Close()
+	}()
+
+	for rows.Next() {
+		var tablePrefix string
+		var tableID int
+		if err := rows.Scan(&tablePrefix, &tableID); err != nil {
+			return fmt.Errorf("scanning table name: %s", err)
+		}
+
+		if err := bs.calculateTreeLeavesForTable(ctx, chainID, tablePrefix, tableID); err != nil {
+			return fmt.Errorf("calculate leaves for table: %s", err)
+		}
+	}
+
+	return nil
+}
+
+func (bs *blockScope) calculateTreeLeavesForTable(
+	ctx context.Context,
+	chainID tableland.ChainID,
+	tablePrefix string,
+	tableID int,
+) error {
+	tableName := fmt.Sprintf("%s_%d_%d", tablePrefix, chainID, tableID)
+	tableRows, err := bs.txn.QueryContext(ctx, fmt.Sprintf("SELECT * FROM %s", tableName))
+	if err != nil {
+		return fmt.Errorf("fetching rows from %s: %s", tableName, err)
+	}
+	defer func() {
+		_ = tableRows.Close()
+	}()
+
+	cols, err := tableRows.Columns()
+	if err != nil {
+		return fmt.Errorf("getting the columns of row: %s", err)
+	}
+
+	rawBuffer := make([]sql.RawBytes, len(cols))
+	scanCallArgs := make([]interface{}, len(rawBuffer))
+	for i := range rawBuffer {
+		scanCallArgs[i] = &rawBuffer[i]
+	}
+
+	leaves := []byte{}
+	for tableRows.Next() {
+		if err := tableRows.Scan(scanCallArgs...); err != nil {
+			return fmt.Errorf("table row scan: %s", err)
+		}
+
+		rowHash := sha3.New256()
+		for _, col := range rawBuffer {
+			rowHash.Write(col)
```

Comment:
From what I understood there it's that it does not follow any standard behavior, but it has a deterministic behavior: 
> it is the current habit of choice of SQLite specifically to return the columns in the order they are created in the internal structure, I assume which is simply because it happens to be the order in which they are parsed, which happens to be the order in which they appear in the CREATE statement

The problem is that they can change that behavior in a future version (because it's not standard) and also ALTER tables can mess up things (we don't support `ALTER TABLE`, at least yet). 

We could sort it but that implies that the client when producing the leaf for verification, would have to sort it too. Maybe that's ok? I think so.